File: docs/src/testers/functions/management/check_run_status.ts
// @sniptest filename=check_run_status.ts
// @sniptest show=10-19
import assert from 'node:assert/strict';

const functionId = process.env.NOTTE_FUNCTION_ID!;

import { NotteClient as FixtureClient } from 'notte-sdk';

const completed = await new FixtureClient().NotteFunction({ function_id: functionId }).run({ url: 'https://example.com' }, { stream: false });
const runId = completed.function_run_id;

import { NotteClient } from 'notte-sdk';

const client = new NotteClient();

// Get run details
const runStatus = await client.NotteFunction({ function_id: functionId }).getRun(runId);

console.log(`Status: ${runStatus.status}`); // "active", "closed", "failed"
console.log('Result:', runStatus.result);
console.log(`Session ID: ${runStatus.session_id}`);

assert.equal(runStatus.status, 'closed');
assert.deepEqual(JSON.parse(runStatus.result!), { url: 'https://example.com', search_query: '' });
export { runStatus as run_status, runId as run_id };

File: docs/src/testers/functions/management/check_run_status.py
# @sniptest filename=check_run_status.py
# @sniptest show=12-21
# ruff: noqa: E402 -- hidden live fixture precedes displayed imports
import os
from json import loads

function_id = os.environ["NOTTE_FUNCTION_ID"]

from notte_sdk import NotteClient as FixtureClient

completed = FixtureClient().Function(function_id).run(url="https://example.com", stream=False)
run_id = completed.function_run_id

from notte_sdk import NotteClient

client = NotteClient()

# Get run details
run_status = client.functions.get_run(function_id, run_id)

print(f"Status: {run_status.status}")  # "active", "closed", "failed"
print(f"Result: {run_status.result}")
print(f"Session ID: {run_status.session_id}")

assert run_status.status == "closed"
assert loads(run_status.result) == {"url": "https://example.com", "search_query": ""}
